runtime/tests/Test_OpenQasmBuilder.cpp

```diff
@@ -0,0 +1,329 @@
+// Copyright 2023 Xanadu Quantum Technologies Inc.
+
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+
+//     http://www.apache.org/licenses/LICENSE-2.0
+
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "openqasm/OpenQasmBuilder.hpp"
+
+#include <catch2/catch.hpp>
+
+using namespace Catalyst::Runtime::Device::OpenQasm;
+
+TEST_CASE("Test lookup openqasm gate names from QIR -> OpenQasm map", "[openqasm]")
+{
+    // Check supported gates
+    CHECK(lookup_qasm_gate_name("PauliX") == "x");
+    CHECK(lookup_qasm_gate_name("PauliY") == "y");
+    CHECK(lookup_qasm_gate_name("PauliZ") == "z");
+    CHECK(lookup_qasm_gate_name("Hadamard") == "h");
+    CHECK(lookup_qasm_gate_name("S") == "s");
+    CHECK(lookup_qasm_gate_name("T") == "t");
+    CHECK(lookup_qasm_gate_name("CNOT") == "cnot");
+    CHECK(lookup_qasm_gate_name("CZ") == "cz");
+    CHECK(lookup_qasm_gate_name("SWAP") == "swap");
+    CHECK(lookup_qasm_gate_name("PhaseShift") == "phaseshift");
+    CHECK(lookup_qasm_gate_name("RX") == "rx");
+    CHECK(lookup_qasm_gate_name("RY") == "ry");
+    CHECK(lookup_qasm_gate_name("RZ") == "rz");
+    CHECK(lookup_qasm_gate_name("CSWAP") == "cswap");
+    CHECK(lookup_qasm_gate_name("PSWAP") == "pswap");
+    CHECK(lookup_qasm_gate_name("ISWAP") == "iswap");
+    CHECK(lookup_qasm_gate_name("Toffoli") == "ccnot");
```

Comment:
Hmm a bit strange of a test no (since we essentially just duplicate the table in the code in the test file)? 😅
Or this just for coverage?

Reply:
This was just for checking the coverage and we don't really need that :+1: